system-addon/content-src/lib/snippets.js

```diff
@@ -316,17 +327,19 @@ function addSnippetsSubscriber(store) {
 
   store.subscribe(async () => {
     const state = store.getState();
-    // state.Snippets.initialized:  Should snippets be initialised?
-    // snippets.initialized:        Is SnippetsProvider currently initialised?
-    if (state.Snippets.initialized && !snippets.initialized &&
-      state.Snippets.onboardingFinished) {
-      // Don't call init multiple times
-      if (!initializing) {
-        initializing = true;
-        await snippets.init({appData: state.Snippets});
-        initializing = false;
-      }
-    } else if (state.Snippets.initialized === false && snippets.initialized) {
+    // state.Prefs.values["feeds.snippets"]:  Should snippets be shown?
+    // state.Snippets.initialized             Is the snippets data initialized?
+    // snippets.initialized:                  Is SnippetsProvider currently initialised?
+    if (state.Prefs.values["feeds.snippets"] &&
+      state.Snippets.initialized &&
+     !snippets.initialized &&
+     // Don't call init multiple times
+     !initializing
+    ) {
+      initializing = true;
+      await snippets.init({appData: state.Snippets});
+      initializing = false;
+    } else if (state.Prefs.values["feeds.snippets"] === false && snippets.initialized) {
```

Comment:
Just noting that we could be `initializing` and forget to `uninit`.

Reply:
Also, this has a behavior change from before where it checked `state.Snippets.initialized` and that could be false when activity stream is shutting down with `UNINIT`. We no longer `SnippetsProvider.uninit` when shutting down, so stuff could be differently broken, but other things are broken with an existing page when the pref is turned off. And soon we wouldn't shut down unless firefox itself is shutting down.